### PR TITLE
Updates createFromParcel to use best classloader for type. fixes #70

### LINKED ISSUE #70: createFromParcel does not use correct ClassLoaders
This is whats being generated:

    @Override
    public AutoValue_Entry createFromParcel(Parcel in) {
      ClassLoader cl = AutoValue_Entry.class.getClassLoader();
      return new AutoValue_Entry(
          in.readString(),
          (EntryAccount) in.readParcelable(cl),
          in.readInt() == 0 ? (EntryLogo) in.readParcelable(cl) : null
      );
    }

The generated method uses the Entry ClassLoader for both objects, but both EntryAccount and EntryLogo should use their own ClassLoader.

This is the exception I get when trying to use the Parcelable in an Intent.
android.os.BadParcelableException: ClassNotFoundException when unmarshalling: AutoValue_Entry
So, I'm guessing this is the problem.

This is my Entry class (I'm using this library alongside auto-value-gson) :

    @AutoValue
    public abstract class Entry implements Parcelable {

        @SerializedName("introduction") abstract public String introduction();
        @SerializedName("account") abstract public EntryAccount account();
        @SerializedName("logo") @Nullable abstract public EntryLogo logo();

        public static TypeAdapter<Entry> typeAdapter(Gson gson) {
            return new AutoValue_Entry.GsonTypeAdapter(gson);
        }

    }

What could be the problem

### PR DESCRIPTION
**Pull request by rharter**
For Map:

* If the type is parameterized, checks the value param for parcelability.
* If value isn't parcelable, checks the key type for parcelability.

Is this necessary?  When parceling a map, do both value and key need to be parcelable?  I'll have to check.

### CODE REVIEW

**Comment by rharter** [comment]
Interesing 

> The Map keys must be String objects.

from [https://developer.android.com/reference/android/os/Parcel.html#writeMap(java.util.Map)](https://developer.android.com/reference/android/os/Parcel.html#writeMap(java.util.Map))

**Comment by rharter** [comment]
So there are some extra restrictions around valid Maps, in the keys need to be strings and values must be valid Parcelable types.  I'm going to open an issue and that can be updated in the `isValidType()` call.

**Comment by JakeWharton** [comment]
I guess that solves the classloader problem for maps as well :+1:

File: auto-value-parcel/src/main/java/com/ryanharter/auto/value/parcel/AutoValueParcelExtension.java [hunk 1]
```diff
@@ -297,9 +296,8 @@ FieldSpec generateCreator(ProcessingEnvironment env, List<Property> properties,
             typeAdapters.get(property.typeAdapter));
       } else {
         final TypeName typeName = Parcelables.getTypeNameFromProperty(property, typeUtils);
-        requiresClassLoader |= Parcelables.isTypeRequiresClassLoader(typeName);
```
Comment (JakeWharton) [inline auto-value-parcel/src/main/java/com/ryanharter/auto/value/parcel/AutoValueParcelExtension.java]: This method is now unused and can be deleted.

**Comment by JakeWharton** [comment]
:+1:

**Pull request merged**